src/kmeans.jl

```diff
@@ -238,49 +236,44 @@ end
 #  (specific to the case where samples are not weighted)
 #
 function update_centers!(
-    x::Matrix{T},                   # in: sample matrix (d x n)
-    w::Nothing,                        # in: sample weights
-    assignments::Vector{Int},       # in: assignments (n)
-    to_update::Vector{Bool},        # in: whether a center needs update (k)
-    centers::Matrix{T},             # out: updated centers (d x k)
-    cweights::Vector) where T<:AbstractFloat               # out: updated cluster weights (k)
+    X::AbstractMatrix{<:Real},       # in: sample matrix (d x n)
+    w::Nothing,                      # in: sample weights
+    assignments::Vector{Int},        # in: assignments (n)
+    to_update::Vector{Bool},         # in: whether a center needs update (k)
+    centers::AbstractMatrix{<:Real}, # out: updated centers (d x k)
+    cweights::Vector{Float64})       # out: updated cluster weights (k)
 
-    d::Int = size(x, 1)
-    n::Int = size(x, 2)
-    k::Int = size(centers, 2)
+    (d, n), k = size(X), size(centers, 2)
 
     # initialize center weights
-    for i = 1 : k
-        if to_update[i]
-            cweights[i] = 0.
-        end
-    end
+    cweights[to_update] .= 0.0
 
     # accumulate columns
-    @inbounds for j = 1 : n
+    @inbounds for j = 1:n
         cj = assignments[j]
         1 <= cj <= k || error("assignment out of boundary.")
+
         if to_update[cj]
             if cweights[cj] > 0
                 for i = 1:d
-                    centers[i, cj] += x[i, j]
+                    centers[i, cj] += X[i, j]
                 end
             else
                 for i = 1:d
-                    centers[i, cj] = x[i, j]
+                    centers[i, cj] = X[i, j]
                 end
             end
             cweights[cj] += 1
         end
     end
 
     # sum ==> mean
-    for j = 1:k
+    @inbounds for j = 1:k
         if to_update[j]
-            @inbounds cj::T = 1 / cweights[j]
-            vj = view(centers,:,j)
+            cj = 1.0 / cweights[j]
```

Comment:
```suggestion
            centers[:, j] .*= 1.0 / cweights[j]
```

Reply:
Should keep `1` instead of `1.0`, as it preserves the type.